src/syscall_mock/mod.rs

```diff
@@ -133,26 +136,45 @@ impl SyscallMock {
         Ok(path)
     }
 
-    pub fn handle_end(mut self, exitcode: i32) -> TestResult {
+    pub fn handle_end(mut self, exitcode: i32, captured_stderr: Option<Vec<u8>>) -> TestResult {
         if let Some(expected_step) = self.protocol.steps.pop_front() {
-            self.register_error(&expected_step.command.format(), "<script terminated>");
+            self.register_step_error(&expected_step.command.format(), "<script terminated>");
         }
         if exitcode != self.protocol.exitcode {
-            self.register_error(
+            self.register_step_error(
                 &format!("<exitcode {}>", self.protocol.exitcode),
                 &format!("<exitcode {}>", exitcode),
             );
         }
+        if let Some(expected_stderr) = &self.protocol.stderr {
+            match captured_stderr {
+                None => panic!("stderr expected, but not captured"),
+                Some(captured_stderr) => {
```

Comment:
not sure if the syntax would allow it, but i believe you could move the `if &captured_stderr != expected_stderr` condition into the pattern match here as a guard.

the benefit would be removing a level of indentation within the block below, but that would also mean adding a `_ => { }` branch at the end so that the arms are comprehensive. Because of that, I'd be happy with either approach here :slightly_smiling_face:

Reply:
Syntactically that'd be allowed. I tried it out and it felt less clear than the current version though.

Reply:
makes sense to me! can already imagine how terse that would look. I'll mark this as resolved :+1: